pgjdbc/src/main/java/org/postgresql/core/v3/SqlSerializationContext.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2025, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.core.v3;
+
+/**
+ * Specifies the properties required to convert SQL to String.
+ */
+public interface SqlSerializationContext {
```

Comment:
@bokken , @sehrope , WDYT of this?
I'm not sure regarding the name, however, I like the interface approach better than two methods `toString` and `toExecutableSql`.

Reply:
I like this idea